### PR TITLE
INT-1836: 1.4.x builds and higher broken on OSX

### PR DESCRIPTION
**Pull request by imlucas**
The primary commit to review is [	INT-1836 🍎 Symlink `Electron` to `${productName}`](https://github.com/mongodb-js/hadron-build/commit/7427843e2cb1d07c3b94093e4362cb9192a76b95).

Verify by running `DEBUG=* npm test -- test/release.test.js` on your macbook.

---
This change is [Reviewable](https://reviewable.io/reviews/mongodb-js/hadron-build/14)

### CODE REVIEW

File: commands/release.js [hunk 1]
```diff
@@ -86,16 +86,35 @@ function createCompileCache(CONFIG, done) {
...
+    /**
+     * @see https://jira.mongodb.org/browse/INT-1836
+     */
+    cli.debug('Ensuring `Contents/MacOS/Electron` is symlinked');
+
+    const cwd = process.cwd();
+
+    process.chdir(path.join(CONFIG.appPath, 'Contents', 'MacOS'));
+
+    fs.ensureSymlink(CONFIG.productName, 'Electron', function(_err) {
```
Comment (durran) [inline commands/release.js]: What does this do on windows? I guess as long as it doesn't break the build we're good here.

Comment (imlucas) [inline commands/release.js]: Yep there is a guard on `CONFIG.platform` so this will only be run on OS X.  Thanks for checking this out!

**Comment by imlucas** [comment]
Published in `hadron-build@0.9.0`

**Comment by imlucas** [comment]
This was first identified and will resolve 10gen/compass#478

**Pull request merged**